src/Components/Card/Card.js

```diff
@@ -33,6 +37,46 @@ const Card = ({ product }) => {
     }
   };
 
+  const handleAddToWishlist = async (product) => {
+    if (authState.isLoggedIn) {
+      try {
+        const responce = await axios.post(
+          "/api/user/wishlist",
+          { product },
+          {
+            headers: { authorization: authState.token },
+          }
+        );
```

Comment:
Suggestion: You can move the `handleAddToWishlist` and `handleRemoveFromWishlist` to the Wishlist context to make the component look cleaner.

Reply:
I am not sure whether we have to it or not. But, I will try